src/map.c

```diff
@@ -9,25 +9,30 @@
 #include "cli/progress.h"
 
 // call must involve i
-SEXP call_loop(SEXP env, SEXP call, int n, SEXPTYPE type, int force_args,
-               SEXP progress) {
-  SEXP i = Rf_install("i");
-  SEXP i_val = Rf_findVarInFrame(env, i);
-
+SEXP call_loop(SEXP env,
+               SEXP call,
+               SEXPTYPE type,
+               SEXP progress,
+               int n,
+               SEXP names,
+               int* p_i,
+               int force) {
   SEXP bar = PROTECT(cli_progress_bar(n, progress));
 
   SEXP out = PROTECT(Rf_allocVector(type, n));
-  SEXP names_val = Rf_findVarInFrame(env, R_NamesSymbol);
-  Rf_setAttrib(out, R_NamesSymbol, names_val);
+  Rf_setAttrib(out, R_NamesSymbol, names);
 
   for (int i = 0; i < n; ++i) {
```

Comment:
The previous overriding of `SEXP i` with the `int i = 0` counter in the loop was killing me 😬